### PR TITLE
layers: Add fix for lx172 to catch invalid line widths.

### PR DESCRIPTION
**Pull request by MarkY-LunarG**
Validate line width in both vkCreateGraphicsPipelines and in
vkCmdSetLineWidth.  Also, add a warning in vkCmdSetLineWidth
if the user calls it but doesn't enable dynamic line width.

Change-Id: I62118da9cb5282fcc22b1506e9be2db82b5f4a40

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -2945,6 +2945,31 @@ static bool validate_and_update_draw_state(layer_data *my_data, GLOBAL_CB_NODE *
     return result;
 }
 
+// Validate HW line width capabilities prior to setting requested line width.
+static VkBool32 verifyLineWidth(layer_data *my_data, DRAW_STATE_ERROR dsError, uint64_t &target, float lineWidth) {
```
Comment (lentinem) [inline layers/core_validation.cpp]: I would use const uint64_t &target.

**Comment by lentinem** [comment]
+1

File: layers/core_validation.cpp (hunk 1 above)
Comment (tobine) [inline layers/core_validation.cpp]: bool return type

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -2945,6 +2945,31 @@ static bool validate_and_update_draw_state(layer_data *my_data, GLOBAL_CB_NODE *
...
+    VkBool32 skipCall = VK_FALSE;
```
Comment (tobine) [inline layers/core_validation.cpp]: bool here and use "skip_call" I'm trying to standardize variable names on lower-case going fwd

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -3102,6 +3127,14 @@ static bool verifyPipelineCreateState(layer_data *my_data, const VkDevice device
                                 pPipeline->graphicsPipelineCI.pTessellationState->patchControlPoints);
         }
     }
+    // If a rasterization state is provided, make sure that the line width conforms to the HW.
+    if (pPipeline->graphicsPipelineCI.pRasterizationState) {
+        VkBool32 dynLineWidth = isDynamic(pPipeline, VK_DYNAMIC_STATE_LINE_WIDTH);
```
Comment (tobine) [inline layers/core_validation.cpp]: switch to "bool" type
"dyn_line_width" preferred

File: layers/core_validation.cpp [hunk 4]
```diff
@@ -6991,6 +7024,17 @@ VK_LAYER_EXPORT VKAPI_ATTR void VKAPI_CALL vkCmdSetLineWidth(VkCommandBuffer com
     if (pCB) {
         skipCall |= addCmd(dev_data, pCB, CMD_SETLINEWIDTHSTATE, "vkCmdSetLineWidth()");
         pCB->status |= CBSTATUS_LINE_WIDTH_SET;
+
+        PIPELINE_NODE *pPipeTrav = getPipeline(dev_data, pCB->lastBound[VK_PIPELINE_BIND_POINT_GRAPHICS].pipeline);
+        VkBool32 dynLineWidth = isDynamic(pPipeTrav, VK_DYNAMIC_STATE_LINE_WIDTH);
```
Comment (tobine) [inline layers/core_validation.cpp]: switch to "bool" type
"dyn_line_width" preferred

**Comment by tobine** [comment]
Cube and Tri are now hitting errors based on this check:

`ERROR: [DS] Code 10 : Attempt to set lineWidth to 0.000000 but physical device wideLines feature not supported/enabled so lineWidth must be 1.0f!`

Can you update those demos so they are no longer flagging these errors?

**Comment by MarkY-LunarG** [comment]
Updated with fixes to demos.  Also, I already updated and committed fixes to VulkanSamples which had several of the same issues.

**Comment by tobine** [comment]
cherry-picked and pushed to master

**Pull request closed without merging**